src/patternfly/components/CodeEditor/code-editor.scss

```diff
@@ -70,62 +87,65 @@
+.#{$code-editor}__header-content {
   display: flex;
-
-  .#{$button}.pf-m-control {
-    --#{$button}--m-control--Color: var(--#{$code-editor}__controls--c-button--m-control--Color);
-
-    &:hover {
-      --#{$code-editor}__controls--c-button--m-control--Color: var(--#{$code-editor}__controls--c-button--m-control--hover--Color);
-    }
-
-    &:focus {
-      --#{$code-editor}__controls--c-button--m-control--Color: var(--#{$code-editor}__controls--c-button--m-control--focus--Color);
-    }
-
-    &:disabled {
-      &::after {
-        border-block-end-color: var(--#{$code-editor}__controls--c-button--m-control--disabled--after--BorderBottomColor);
-      }
-    }
+  flex: 1;
+  align-items: flex-end;
+  padding-block-start: var(--#{$code-editor}__header-content--PaddingBlockStart);
+  padding-block-end: var(--#{$code-editor}__header-content--PaddingBlockEnd);
+  padding-inline-start: var(--#{$code-editor}__header-content--PaddingInlineStart);
+  padding-inline-end: var(--#{$code-editor}__header-content--PaddingInlineEnd);
+  background-color: var(--#{$code-editor}__header-content--BackgroundColor);
+  border-radius: var(--#{$code-editor}__tab--BorderRadius) 0 0 0;
+
+  &.pf-m-plain {
+    --#{$code-editor}__header-content--BackgroundColor: var( --#{$code-editor}__header-content--m-plain--BackgroundColor);
   }
 }
 
+.#{$code-editor}__controls {
+  display: flex;
+  gap: var(--#{$code-editor}__controls--Gap);
+  align-items: flex-start;
+}
+
 .#{$code-editor}__header-main {
   flex-grow: 1;
+  align-self: center;
   padding-inline-start: var(--#{$code-editor}__header-main--PaddingLeft);
   padding-inline-end: var(--#{$code-editor}__header-main--PaddingRight);
 }
 
+.#{$code-editor}__upload {
+  position: relative;
+  padding-block-start: var(--#{$code-editor}__upload--PaddingBlockStart);
+  padding-block-end: var(--#{$code-editor}__upload--PaddingBlockEnd);
+  padding-inline-start: var(--#{$code-editor}__upload--PaddingInlineStart);
+  padding-inline-end: var(--#{$code-editor}__upload--PaddingInlineEnd);
+
+  &::before {
+    position: absolute;
+    inset-block-start: var(--#{$code-editor}__upload--PaddingBlockStart);
+    inset-block-end: var(--#{$code-editor}__upload--PaddingBlockEnd);
+    inset-inline-start: var(--#{$code-editor}__upload--PaddingInlineStart);
+    inset-inline-end: var(--#{$code-editor}__upload--PaddingInlineEnd);
+    content: "";
+    border: var(--#{$code-editor}__upload--before--BorderWidth) dashed var(--#{$code-editor}__upload--before--BorderColor);
+    border-radius: var(--#{$code-editor}__upload--BorderRadius);
+  }
+}
+
 .#{$code-editor}__main {
   position: relative;
   color: var(--#{$code-editor}__main--Color, inherit);
   background-color: var(--#{$code-editor}__main--BackgroundColor);
   border: var(--#{$code-editor}__main--BorderWidth) solid;
   border-color: var(--#{$code-editor}__main--BorderColor);
+  border-radius: 0 0 var(--#{$code-editor}__tab--BorderRadius) var(--#{$code-editor}__tab--BorderRadius);
```

Comment:
Looks like we have `__tab--BorderRadius` defined on `__main`, `__header-content`, and `__tab`. That means to change any of them, a user will set `--#{$code-editor}__tab--BorderRadius`.

If we decide to give main (as an example) a unique border-radius via `--#{$code-editor}__main--BorderRadius`, then `--#{$code-editor}__tab--BorderRadius` will no longer control `__main`'s border-radius. If a user themed  `--#{$code-editor}__tab--BorderRadius`, that will no longer style `__main`, which is technically a breaking change. 

That logic and what we consider a breaking change may change as we nail down the definition of what what a breaking change is in core now that we have such a rich set of vars to style everything, but for now would you mind creating unique vars for `__main`, `__header-content`, and `__tab`?